### PR TITLE
Use solr for collection and unit's `items` endpoints

### PR DESCRIPTION
**Pull request by masaball**
Related issue: #6944

### CODE REVIEW

File: spec/controllers/admin_collections_controller_spec.rb [hunk 1]
```diff
@@ -238,6 +238,17 @@
...
     end
+
+    context 'read from solr' do
+      it 'should not read from fedora' do
+        collection = FactoryBot.create(:collection, items: 5)
+        WebMock.reset_executed_requests!
+        get :index
+        get :items, params: { id: collection.id, format: 'json', per_page: '2' }
+        expect(a_request(:post, /#{ActiveFedora.solr.conn.uri.to_s}/)).to have_been_made.at_least_once
+        expect(a_request(:post, /#{ActiveFedora.solr.conn.uri.to_s}.*\?fl=id&q=/)).not_to have_been_made
```
Comment (cjcolvar) [inline spec/controllers/admin_collections_controller_spec.rb]: I tested this out locally and it looks like the MediaObject proxy is reifying due to `avalon_resource_type` being called.  Since the proxies in the test don't have any sections this value doesn't get set and stored in solr leading to the reification.  If we default avalon_resource_type to nil in `config/initializers/presenter_config.rb` then I think it won't reify and this test can be changed to the stricter version used in the admin units controller spec.

This helped chase down another edge case where reification is happening so it is really helpful in our quest to avoid calling fedora from the front end!

**Review by cjcolvar — COMMENTED** [review]
This looks good!  I would have expected that the switch to speedyaf proxies would have caused problems with `paginate` so glad that it was easy and straight-forward!
